app/reducers/announcements.js

```diff
@@ -4,37 +4,32 @@ import { createSelector } from 'reselect';
 import { Announcements } from '../actions/ActionTypes';
 import createEntityReducer from 'app/utils/createEntityReducer';
 import moment from 'moment-timezone';
+import produce from 'immer';
+
+type State = any;
 
 export default createEntityReducer({
   key: 'announcements',
   types: {
     fetch: Announcements.FETCH_ALL,
     mutate: Announcements.CREATE
   },
-  mutate(state, action) {
-    switch (action.type) {
-      case Announcements.SEND.SUCCESS: {
-        const { announcementId } = action.meta;
-        return {
-          ...state,
-          byId: {
-            ...state.byId,
-            [announcementId]: {
-              ...state.byId[announcementId],
-              sent: moment()
-            }
-          }
-        };
+  mutate: (state: State, action): State =>
+    produce(
```

Comment:
Will `produce` return the state if none of the cases trigger, since you don't write any default case?

Reply:
Yes, it returns newState, no matter how/if you mutate it.